crates/storage/provider/src/providers/snapshot/manager.rs

```diff
@@ -35,9 +36,12 @@ use tokio::sync::watch;
 /// - `HashMap<SnapshotSegment, BTreeMap<TxNumber, RangeInclusive<BlockNumber>>>`
 type SegmentRanges = HashMap<SnapshotSegment, BTreeMap<u64, RangeInclusive<u64>>>;
 
-/// [`SnapshotProvider`] manages all existing [`SnapshotJarProvider`].
+/// Type alias for the shared snapshot provider, wrapping the inner implementation.
+pub type SnapshotProvider = Arc<SnapshotProviderInner>;
```

Comment:
personally, I really dislike aliases for these things, because it makes it a lot harder to follow.

I'd rather have this as a new type struct

Reply:
No problem let me know we can do that as a new structure with only the arc inside. Do you prefer that?

Reply:
@mattsse I've just added a structure for `SnapshotProvider`, let me know if it is better for you.